pkg/nitro_enclaves_cpu_plugin/device_plugin.go

```diff
@@ -0,0 +1,369 @@
+// Copyright 2025 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package nitro_enclaves_cpu_plugin
+
+import (
+	"errors"
+	"fmt"
+	"google.golang.org/grpc/credentials/insecure"
+	"k8s-ne-device-plugin/pkg/config"
+	"net"
+	"os"
+	"os/signal"
+	"path"
+	"strconv"
+	"strings"
+	"syscall"
+	"time"
+
+	"github.com/golang/glog"
+	"golang.org/x/net/context"
+	"google.golang.org/grpc"
+	pluginapi "k8s.io/kubelet/pkg/apis/deviceplugin/v1beta1"
+)
+
+const (
+	deviceName                     = "nitro_enclaves_cpus"
+	devicePluginServerReadyTimeout = 10
+	deviceOfflineCPUsPath          = "/sys/devices/system/cpu/offline"
+)
+
+var cpuIdCounter = 0
+
+func (necdp *NitroEnclavesCPUDevicePlugin) socketPath() string {
+	return pluginapi.DevicePluginPath + deviceName + ".sock"
+}
+
+func (necdp *NitroEnclavesCPUDevicePlugin) resourceName() string {
+	return "aws.ec2.nitro/" + deviceName
+}
+
+type IBasicDevicePlugin interface {
+	Start() error
+	Stop()
+	socketPath() string
+	resourceName() string
+}
+
+// NitroEnclavesCPUDevicePlugin implements the Kubernetes device plugin API
+type NitroEnclavesCPUDevicePlugin struct {
+	devices []*pluginapi.Device
+
+	stop chan interface{}
+
+	server *grpc.Server
+	pluginapi.DevicePluginServer
+	IBasicDevicePlugin
+}
+
+// FileReader interface defines the behavior for reading CPU information
+type fileReader interface {
+	read() (string, error)
+}
+
+// SystemFileReader implements FileReader for the actual system
+type systemFileReader struct {
+	path string
+}
+
+func newSystemFileReader(path string) *systemFileReader {
+	return &systemFileReader{path: path}
+}
+
+func (s *systemFileReader) read() (string, error) {
```

Comment:
`fileReader` interface and `systemFileReader` struct look redundant here.
Even this function content can be used directly in place where it is needed.